src/machine/machine_atsamd51p20.go

```diff
@@ -0,0 +1,70 @@
+// +build sam,atsamd51,atsamd51p20
+
+// Peripheral abstraction layer for the atsamd51.
+//
+// Datasheet:
+// http://ww1.microchip.com/downloads/en/DeviceDoc/60001507C.pdf
+//
+package machine
+
+import "device/sam"
+
+const HSRAM_SIZE = 0x00040000
+
+// InitPWM initializes the PWM interface.
+func InitPWM() {
+	// turn on timer clocks used for PWM
+	sam.MCLK.APBBMASK.SetBits(sam.MCLK_APBBMASK_TCC0_ | sam.MCLK_APBBMASK_TCC1_)
+	sam.MCLK.APBCMASK.SetBits(sam.MCLK_APBCMASK_TCC2_ | sam.MCLK_APBCMASK_TCC3_)
+	sam.MCLK.APBDMASK.SetBits(sam.MCLK_APBDMASK_TCC4_)
+
+	//use clock generator 0
+	sam.GCLK.PCHCTRL[sam.PCHCTRL_GCLK_TCC0].Set((sam.GCLK_PCHCTRL_GEN_GCLK0 << sam.GCLK_PCHCTRL_GEN_Pos) |
+		sam.GCLK_PCHCTRL_CHEN)
+	sam.GCLK.PCHCTRL[sam.PCHCTRL_GCLK_TCC2].Set((sam.GCLK_PCHCTRL_GEN_GCLK0 << sam.GCLK_PCHCTRL_GEN_Pos) |
+		sam.GCLK_PCHCTRL_CHEN)
+	sam.GCLK.PCHCTRL[sam.PCHCTRL_GCLK_TCC4].Set((sam.GCLK_PCHCTRL_GEN_GCLK0 << sam.GCLK_PCHCTRL_GEN_Pos) |
+		sam.GCLK_PCHCTRL_CHEN)
+}
+
+// getTimer returns the timer to be used for PWM on this pin
+func (pwm PWM) getTimer() *sam.TCC_Type {
```

Comment:
The same code should be used for ATSAMD51P20 and ATSAMD51P19 (wioterminal), except that the ROM/RAM is different.
So the only difference in machine_atsamd51pXX.go will be HSRAM_SIZE.
I think `PWM.getTime()` should include both.
I think it would be better to keep the definitions that exist in `machine_atsamd51p19.go` .

https://github.com/tinygo-org/tinygo/blob/release/src/machine/machine_atsamd51p19.go#L30

Reply:
`getTimer` appears different between `atsamd15p19.go` (Wio) and `atsamd51p20.go` (GC). For example, `PA16` is defined on the P19, but it is _not_ defined on the P20; the P20 would return a nil timer for `PA16`.

Note that I've based these mappings for the P20 on the Arduino core ([see here](https://github.com/adafruit/ArduinoCore-samd/blob/aebf3cb686daa2bb8ee8e85febf6ed656241d933/variants/grand_central_m4/variant.cpp#L29)), so it is possible their pin definitions are not the only permissable configuration; i.e., the P19 implementation of `getTimer` might be valid for the P20 as well (I have no idea). I just wanted to point out they are currently not equivalent.

If this is the case, though, I think the J20 is a closer match to the P20, since it also has an equal value for `HSRAM_SIZE`, but otherwise has an identical implementation of `getTimer` to the P19.

### Different PR 

However, I'm not sure this effort should be restricted to these two variants, and thus should not be part of this PR. For example, all of the following source files are identical (except for `HSRAM_SIZE`):

- `machine_atsamd51j19.go`
- `machine_atsamd51j20.go`
- `machine_atsamd51p19.go`
- `machine_atsamd51p20.go` (if its pins can use a different configuration than implemented in this PR)

I propose that all of these files be combined into a single source file.

Furthermore, since `HSRAM_SIZE` can be derived from the linker script for each of this chips, I would recommend we not redefine here in the source file, and let the linker script define it. Then this common source file can just use a `//go:extern HSRAM_SIZE` declaration to get the appropriate value.

Reply:
Yes, let's try to handle it with different PR.

It will probably need to be set for each Pin Count in GJNP. However, if we define a dummy pin (PCxx / PDxx) that does not exist in the G series, we may be able to use the same source. However, in that case, it will be difficult to check for errors.

Reply:
So, I think this part is fine as it is now.
Thank you for your comments.